### PR TITLE
Add itch.app v17.6.0

### PR DESCRIPTION
**Pull request by optikfluffel**
- [x] Checked there aren’t open [pull requests](https://github.com/caskroom/homebrew-cask/pulls) for the same cask.
- [x] Checked there aren’t closed [issues](https://github.com/caskroom/homebrew-cask/issues) where that cask was already refused.
- [x] When naming the cask, followed the [token reference](https://github.com/caskroom/homebrew-cask/blob/master/doc/cask_language_reference/token_reference.md).
- [x] Commit message includes cask’s name.
- [x] `brew cask audit --download {{cask_file}}` is error-free.
- [x] `brew cask style --fix {{cask_file}}` left no offenses.
- [x] `brew cask install {{cask_file}}` worked successfully.
- [x] `brew cask uninstall {{cask_file}}` worked successfully.

### CODE REVIEW

File: Casks/itch.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+cask 'itch' do
+  version 'v17.6.0'
```
Comment (vitorgalvao) [inline Casks/itch.rb]: Remove the `v` here.

File: Casks/itch.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+  sha256 'f5ec3f4c9c45a4811f11b79962d84c1e12ac0c42458653c68c02051f555c6671'
+
+  url "https://github.com/itchio/itch/releases/download/#{version}/itch-mac.dmg"
```
Comment (vitorgalvao) [inline Casks/itch.rb]: And add it here.

Comment (vitorgalvao) [inline Casks/itch.rb]: Needs a `# github.com/itchio/itch was verified as official when first introduced to the cask` comment above this line ([example](https://github.com/caskroom/homebrew-cask/blob/4ad02c9f087deb9e553ab0129c079d73c474add8/Casks/0ad.rb#L5); [reasoning](https://github.com/caskroom/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/url.md#when-url-and-homepage-hostnames-differ-add-a-comment)).

Comment (vitorgalvao) [inline Casks/itch.rb]: Please add an [`appcast`](https://github.com/caskroom/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/appcast.md).

**Comment by optikfluffel** [comment]
@vitorgalvao ✅

**Pull request merged**